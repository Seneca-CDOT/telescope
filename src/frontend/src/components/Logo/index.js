import Logo from './Logo.jsx';

export default Logo;
