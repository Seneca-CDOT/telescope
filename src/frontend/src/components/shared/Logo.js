import React from 'react';
import PropTypes from 'prop-types';

const Logo = ({ className, logo }) => (
  <div className={`${className}__logo`}>
    <img src={logo} alt="" />
  </div>
);

Logo.propTypes = {
  className: PropTypes.string,
  logo: PropTypes.any,
};

export default Logo;
