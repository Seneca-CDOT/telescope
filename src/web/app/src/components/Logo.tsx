const logoUrl = '/logo.svg';
const logoBadgedUrl = '/logo-badged.svg';

type LogoProps = {
  height: number;
  width: number;
};

const Logo = ({ height, width }: LogoProps) => (
  <img src={logoUrl} alt="Telescope Logo" height={height} width={width} />
);

export default Logo;
export { logoUrl, logoBadgedUrl };
