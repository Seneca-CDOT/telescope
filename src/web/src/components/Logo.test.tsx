import TestRenderer from 'react-test-renderer';
import Logo from './Logo';

it('renders correctly', () => {
  const tree = TestRenderer.create(<Logo height={50} width={50} />).toJSON();
  expect(tree).toMatchSnapshot();
});
