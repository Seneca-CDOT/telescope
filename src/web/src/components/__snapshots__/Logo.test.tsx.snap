// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt="Telescope Logo"
  height={50}
  src="/logo.svg"
  width={50}
/>
`;
